lib/bolt/cli.rb

```diff
@@ -390,6 +390,14 @@ def handle_parser_errors
     def execute(options)
       if options[:mode] == 'plan' || options[:mode] == 'task'
         begin
+          if Gem.win_platform?
```

Comment:
Should this be `!!File::ALT_SEPARATOR` per https://github.com/puppetlabs/bolt/pull/241#discussion_r162177318?

Reply:
I guess so? I don't understand @Iristyle's comment because Gem is shipped with Ruby now.

Reply:
Hmm good point, this is probably fine then?

Reply:
it's also the logic that net:ssh is using

Reply:
For Windows checking I prefer to not rely on anything external / versioned outside of core Ruby, especially when that code is designed to be mocked, when there's not a clear reason for it to be.

`FILE::ALT_SEPARATOR` is only compiled into Windows versions of Ruby and makes for a fast, simple and well understood (albeit a bit cryptic) check.

`Gem.win_platform?` at https://github.com/rubygems/rubygems/blob/master/lib/rubygems.rb#L1087-L1094 has indeed been around for a long time - originally added in 2007 at  https://github.com/rubygems/rubygems/commit/776d791c55e6d033bc7b450186821edbe4add579 ... but it's a bit more complex, and is also in place for the sake of tests pretending they're on Windows when they're not.

Either should work here, I just prefer not to use helpers on `Gem` or `Bundler` for this.